### PR TITLE
Add update procedure

### PR DESCRIPTION
**Pull request by ruimarinho**
Documents the update procedure as it can be tricky.

R=@JerrySievert

### CODE REVIEW

**Comment by ruimarinho** [comment]
@JerrySievert I can get a reproducible crash when downgrading from `plv8 2.0.0-dev` to `plv8 1.5.3`. It does not happen when upgrading. 

Steps:
1. Install plv8 2.0.0-dev.
2. Execute `DO $$ plv8.elog(WARNING, plv8.version) $$ LANGUAGE plv8;`, confirm output is `2.0.0-dev`. Keep the psql connection/process alive.
3. Copy `plv8.so` and `plv8.control` to destination dirs.
4. Execute `DO $$ plv8.elog(WARNING, plv8.version) $$ LANGUAGE plv8;` again. Output is:

```
ERROR:  unrecognized function API version: -2138417390
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
The connection to the server was lost. Attempting reset: Failed.
```

File: doc/plv8.md [hunk 1]
```diff
@@ -478,6 +479,71 @@ visible from any subsequent function as global variables.
...
+value of the new control file, which may not represent the current PL/v8 version
+in memory.
+
+The best way of finding out which PL/v8 version you're running is by executing:
+
+```sql
+DO $$ plv8.elog(WARNING, plv8.version) $$ LANGUAGE plv8;
+```
+
+Even when using PL/v8 2.0.0, `SELECT plv8_version();` is only indicative of the
```
Comment (ruimarinho) [inline doc/plv8.md]: @JerrySievert can you confirm if the `ALTER EXTENSION` syntax forces some kind of binary reload that I may have missed?

**Comment by ruimarinho** [comment]
@JerrySievert are you still interested in this PR? I don't mind keeping it open, just let me know if you need time to review it.

**Comment by bendiy** [comment]
@ruimarinho I've made some extensive changes to this file in https://github.com/plv8/plv8/pull/198 It might cause a merge conflict for you, but it should be easy to cleanup if @JerrySievert merges it.

File: doc/plv8.md [hunk 2]
```diff
@@ -22,6 +22,7 @@ Implemented features are as follows.
 - Remote debugger
 - Runtime environment separation across users in the same session
 - Start-up procedure
+- Update procedure
```
Comment (bendiy) [inline doc/plv8.md]: Turn this into a link like:
https://github.com/plv8/plv8/pull/204/files#diff-f7db76a4bc8765fece9d587798ae00d7R27
```
- [Update procedure](#update-procedure)
```

Add it to the `README.md` TOC too:
https://github.com/plv8/plv8/pull/204/files#diff-04c6e90faac2675aa89e2176d2eec7d8R28

File: doc/plv8.md [hunk 3]
```diff
@@ -478,6 +479,74 @@ visible from any subsequent function as global variables.
 Remember CREATE FUNCTION also starts the plv8 runtime environment, so make sure
 to SET this GUC before any plv8 actions including CREATE FUNCTION.
 
+Update procedure
+------------------
```
Comment (bendiy) [inline doc/plv8.md]: Convert header format to:
```
## Update procedure
```

**Comment by ruimarinho** [comment]
All done @JerrySievert, @bendiy.

**Pull request merged**